### PR TITLE
Modbus tcp rtu germanbluefox

### PR DESCRIPTION
**Pull request by BauchBeinePoe**
Everything concerning the modbus tcp rtu feature.

### CODE REVIEW

File: src/modbus-tcp-client.js [hunk 1]
```diff
@@ -92,41 +93,40 @@ module.exports = stampit()
...
 
                 // 1. extract mbap
 
-                var mbap    = data.slice(cnt, cnt + 7),
-                    id      = mbap.readUInt16BE(0),
-                    len     = mbap.readUInt16BE(4);
+                var id  = buffer.readUInt16BE(0),
+                    len = buffer.readUInt16BE(4);
 
-                if (id === trashRequestId) {
```
Comment (BauchBeinePoe) [inline src/modbus-tcp-client.js]: Why is this removed? trashed request ids is for requests that have been trashed due to timeouts or stuff. This way they can be identified.

**Comment by BauchBeinePoe** [comment]
I see. I will make an extra pull request for the handling of incoming data.

**Comment by BauchBeinePoe** [comment]
@GermanBluefox You are now a collaborator and push to branches (except the master branch). It would be goodto have a test case for the issue with chopped data packets.

**Comment by GermanBluefox** [comment]
> It would be goodto have a test case for the issue with chopped data packets.
I will make one. But first please merge all my mess. :)

**Comment by BauchBeinePoe** [comment]
Processing of the incomming data for the modbus-tcp-rtu-client and the onData function for the serial-client need some tests.

**Comment by GermanBluefox** [comment]
I will write the test code for it, _**when it will be merged**_ and I will start with the implementation of TCP-RTU Server.
Just now it is too much for me to have 4 changing branches and to switch between them. It is enough for me with my 112 ioBroker repositories and I must jump between them too.

**Comment by GermanBluefox** [comment]
I still want to finish it :) Just get a bit more time and will do that.

**Comment by roccomuso** [comment]
+1
